floyd/client/files.py

```diff
@@ -114,9 +118,33 @@ def sizeof_fmt(num, suffix='B'):
     return "%.1f%s%s" % (num, 'Yi', suffix)
 
 
+def warn_purge_exit(info_msg, filename, exit_msg):
+    """
+    Warn the user that's something went wrong,
+    remove the tarball and provide an exit message.
+    """
+    floyd_logger.info(info_msg)
+    rmtree(os.path.dirname(filename))
+    sys.exit(exit_msg)
+
+
 def create_tarfile(source_dir, filename="/tmp/contents.tar.gz"):
     """
     Create a tar file with the contents of the current directory
     """
-    with tarfile.open(filename, "w:gz") as tar:
-        tar.add(source_dir, arcname=os.path.basename(source_dir))
+    try:
+        # Define the default signal handler for catching: Ctrl-C
+        signal.signal(signal.SIGINT, signal.default_int_handler)
+        with tarfile.open(filename, "w:gz") as tar:
+            tar.add(source_dir, arcname=os.path.basename(source_dir))
+
+    except OSError:  # OSError: [Errno 13] Permission denied
+        warn_purge_exit(info_msg="Permission denied. Removing compressed data...",
```

Comment:
you need to check to make sure it's actually `errno.EPERM` here. there are other types of OSErrors that can happen.